### DISCUSSION TITLE
Ability to reserve a device

### DISCUSSION THREAD

**Discussion by kumareshind** [nautobot/nautobot#1033] [Ideas]
Today, a reservation is viewed from a "rack" perspective. This "rack wise" reservation also restricts devices within a single rack alone (Users can reserve specific units within a rack for future use. An arbitrary set of units within a rack can be associated with a single reservation, **but reservations cannot span multiple racks**. A description is required for each reservation, reservations may optionally be associated with a specific tenant).

I think it will be really good to reserve an arbitrary set of devices across different racks also.

**Comment by kumareshind** [comment]
Thinking more about this, "Reservation" could be a first class model and we should be able to apply this to any entity as a relationship. For example: What if, someone reserve an interface in a switch? Then we need to again think of it as a separate way. Instead, "Reservation" is a model with "duration", "owner", "status", "from", "to", "auto_release"  as its fields (might be more based on more discussions). I can then be able to apply this to a Device or a Rack or an Interface, etc.

**Comment by mzbroch** [comment]
Related to #1047 (dynamic groups / reservation pools)

Reservation idea sounds interesting. Indeed rack reservations serve a different purpose and a more general models are needed to support this feature.

**Reply by kumareshind** [reply]
Anyone interested in this topic?

**Reply by mzbroch** [reply]
At NTC we have started the planning work on this feature. We will keep You posted here on the progress.

**Reply by kumareshind** [reply]
@mzbroch any new news on this? If interested, I can participate in the discussion and possibly contribute. Thanks.

**Reply by mzbroch** [reply]
We made a good progress in the development, that being said core and models have been already created. However, work is in progress for some of the remaining integrations.

**Reply by itdependsnetworks** [reply]
We should have something in a month, I believe we will be able to release publicly, but ping me if you don't hear anything.

**Reply by kumareshind** [reply]
Hi @itdependsnetworks , @mzbroch  - any new updates on this? Thanks a lot..

**Reply by mzbroch** [reply]
Pinging @itdependsnetworks :-)
